hack/migrate-to-antrea.sh

```diff
@@ -0,0 +1,256 @@
+        --antrea-manifest <PATH>    Path to the Antrea manifest file, if not specified, the latest manifest will be used.
+"
+
+_calico_warning_message="
+WARNING: You are trying to migrate from Calico to Antrea. Please NOTE:
+If you are using Kubernetes origin Networkpolicy, all the network policies will still work
+after migration.
+During the migration, the workloads will be restarted and the network will be interrupted.
+If you are using Calico NetworkPolicy or Calico GlobalNetworkPolicy, only PART of
+the network policies could be converted to Antrea NetworkPolicy and Antrea ClusterNetworkPolicy.
+Please check the migration document for details."
+
+_flannel_warning_message="
+WARNING: You are trying to migrate from Flannel to Antrea. Please NOTE:
+During the migration, the workloads will be restarted and the network will be interrupted.
+"
+
+function print_usage {
+    echo "$_usage"
+}
+
+FROM=""
+NETWORK_POLICY=false
+DRY_RUN=false
+BACKUP=false
+OUTPUT_DIR=$(pwd)
+KUBECONFIG="$HOME/.kube/config"
+PLATFORM=$(uname | tr '[:upper:]' '[:lower:]')
+CALICO_MANIFEST="https://docs.projectcalico.org/manifests/calico.yaml"
+FLANNEL_MANIFEST="https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml"
+ANTREA_MANIFEST="./build/yamls/antrea.yml"
+
+while [[ $# -gt 0 ]]; do
+    key="$1"
+
+    case $key in
+    --from)
+        FROM="$2"
+        shift 2
+        ;;
+    --networkpolicy)
+        NETWORK_POLICY=true
+        shift
+        ;;
+    --dry-run)
+        DRY_RUN=true
+        shift
+        ;;
+    --backup)
+        BACKUP=true
+        shift
+        ;;
+    --output-dir)
+        OUTPUT_DIR="$2"
+        shift 2
+        ;;
+    --kubeconfig)
+        KUBECONFIG="$2"
+        shift 2
+        ;;
+    --calico-manifest)
+        CALICO_MANIFEST="$2"
+        shift 2
+        ;;
+    --flannel-manifest)
+        FLANNEL_MANIFEST="$2"
+        shift 2
+        ;;
+    --antrea-manifest)
+        ANTREA_MANIFEST="$2"
+        shift 2
+        ;;
+    -h | --help)
+        print_usage
+        exit 0
+        ;;
+    *) # unknown option
+        echo "Unknown option $1"
+        exit 1
+        ;;
+    esac
+done
+
+if [ "$FROM" == "calico" ]; then
+    echo "$_calico_warning_message"
+elif [ "$FROM" == "flannel" ]; then
+    echo "$_flannel_warning_message"
+else
+    echo "Unsupported CNI: $FROM"
+    exit 1
+fi
+
+echo "Input 'yes' to continue"
+read -r input
+if [ "$input" != "yes" ]; then
+    echo "========== Aborting =========="
+    exit 1
+fi
+
+echo "========== Building antctl =========="
+if [ $DRY_RUN == true ]; then
+    echo "make antctl"
+else
+    make antctl
+    mv ./bin/antctl-${PLATFORM} ./bin/antctl
+fi
+
+echo "========== Checking the environment =========="
+if [ $DRY_RUN == true ]; then
+    echo "./bin/antctl migrate precheck --from $FROM" --networkpolicy=$NETWORK_POLICY
```

Comment:
We can't assume the user has the antrea repo.
It's not right to assume antctl will be located at `./bin` if we will let user to download the file first.

Reply:
I added a check for antctl command, if it does not exist, the script will tell users to download it or build from source code.